File: modules/body/test/matlab/LandmarksTranslatorTest.m
classdef LandmarksTranslatorTest < matlab.unittest.TestCase

    methods (Test)
    
        function convert(testCase)
            translator = ma.body.LandmarksTranslator('LandmarksTranslator',{ ...
              {'LFHD', 'L.HF'}, ...
              {'LBHD', 'L.HB'}, ...
              {'RFHD', 'R.HF'}, ...
              {'RBHD', 'R.HB'}, ...
              {'C7'  , 'C7'  }, ...
            });
            testCase.verifyEqual(translator.convert('LFHD'), 'L.HF');
            testCase.verifyEqual(translator.convert('LBHD'), 'L.HB');
            testCase.verifyEqual(translator.convert('RFHD'), 'R.HF');
            testCase.verifyEqual(translator.convert('RBHD'), 'R.HB');
            testCase.verifyEqual(translator.convert('C7'), 'C7');
            testCase.verifyEqual(translator.convert('RBHD_'), 'RBHD_');
            testCase.verifyEqual(translator.convert('T10'), 'T10');
        end
  
        function convertIfExists(testCase)
            translator = ma.body.LandmarksTranslator('LandmarksTranslator',{
              {'LFHD', 'L.HF'}, ...
              {'LBHD', 'L.HB'}, ...
              {'RFHD', 'R.HF'}, ...
              {'RBHD', 'R.HB'}, ...
              {'C7'  , 'C7'  }, ...
            });
            testCase.verifyEqual(translator.convertIfExists('LFHD'), 'L.HF');
            testCase.verifyEqual(translator.convertIfExists('LBHD'), 'L.HB');
            testCase.verifyEqual(translator.convertIfExists('RFHD'), 'R.HF');
            testCase.verifyEqual(translator.convertIfExists('RBHD'), 'R.HB');
            testCase.verifyEqual(translator.convertIfExists('C7'), 'C7');
            testCase.verifyEqual(isempty(translator.convertIfExists('RBHD_')), true);
            testCase.verifyEqual(isempty(translator.convertIfExists('T10')), true);
        end
        
    end
    
end

File: modules/body/test/python/landmarkstranslatorTest.py
import ma.body
import unittest

class LandmarksTranslatorTest(unittest.TestCase):
  def test_convert(self):
    translator = ma.body.LandmarksTranslator('LandmarksTranslator',[
      ['LFHD', 'L.HF'],
      ['LBHD', 'L.HB'],
      ['RFHD', 'R.HF'],
      ['RBHD', 'R.HB'],
      ['C7'  , 'C7'  ],
    ])
    self.assertEqual(translator.convert('LFHD'), 'L.HF')
    self.assertEqual(translator.convert('LBHD'), 'L.HB')
    self.assertEqual(translator.convert('RFHD'), 'R.HF')
    self.assertEqual(translator.convert('RBHD'), 'R.HB')
    self.assertEqual(translator.convert('C7'), 'C7')
    self.assertEqual(translator.convert('RBHD_'), 'RBHD_')
    self.assertEqual(translator.convert('T10'), 'T10')
  
  def test_convert_if_exists(self):
    translator = ma.body.LandmarksTranslator('LandmarksTranslator',[
      ['LFHD', 'L.HF'],
      ['LBHD', 'L.HB'],
      ['RFHD', 'R.HF'],
      ['RBHD', 'R.HB'],
      ['C7'  , 'C7'  ],
    ])
    self.assertEqual(translator.convertIfExists('LFHD'), 'L.HF')
    self.assertEqual(translator.convertIfExists('LBHD'), 'L.HB')
    self.assertEqual(translator.convertIfExists('RFHD'), 'R.HF')
    self.assertEqual(translator.convertIfExists('RBHD'), 'R.HB')
    self.assertEqual(translator.convertIfExists('C7'), 'C7')
    self.assertEqual(translator.convertIfExists('RBHD_'), '')
    self.assertEqual(translator.convertIfExists('T10'), '')